### ISSUE TITLE
can we make outbound an optional in module net-vpc DNS block

### ISSUE THREAD

**Issue by nagendrasidgs** [GoogleCloudPlatform/cloud-foundation-fabric#821]
https://github.com/GoogleCloudPlatform/cloud-foundation-fabric/blob/master/modules/net-vpc/main.tf 
Line 98

We are trying to enable DNS logs using the net-vpc. But the DNS block is Enforcing the Outbound nameservers to be passed. At a high level, we are peering GCP project DNS with private service connect ApigeeX DNS.

variable "dns_policy" {
  description = "DNS policy setup for the VPC."
  type = object({
    inbound = bool
    logging = bool
    outbound = object({
      private_ns = list(string)
      public_ns  = list(string)
    })
  })
  default = null
}

We are not able to pass the empty list string for private_ns & public_ns. Any recommendations?

Basically, GCP is not enforcing in adding one of the alternatives but the module is.  We don't have any nameserver at this point for internal is there a way to make it optional? If we are manually undoing the nameserver the whole flow works but to keep the terraform state intact we need to disable the outbound.

below is the error:
│ Error: Insufficient target_name_servers blocks
│ 
│   on .terraform/modules/vpc/modules/net-vpc/main.tf line 99, in resource "google_dns_policy" "default":
│   99:     content {
│ 
│ At least 1 "target_name_servers" blocks are required.

**Comment by ludoo** [comment]
If you set `outbound` to `null` it should do what you want, if I understand your use case correctly.

**Comment by nagendrasidgs** [comment]
Are you suggesting something as below:

  dns_policy = {
    inbound  = true
    logging  = true
    outbound = null
  }

**Comment by ludoo** [comment]
Great! Thanks for confirming.
